### PR TITLE
Hide elements

### PR DESCRIPTION
**Pull request by EnderSmith**
This is really a refactor of my previously [unresolved PR](https://github.com/EnderSmith/EnderSmith.github.io/pull/31)
Everything approved here should be ready to pull to master, so expect a quick turn around.
Since then:

`content.json`:
- RENAMED file `content.json` to `preloadedSaveItems.json`
- REMOVED `content` object and all references (including `contentStatus` property)

`diceRoll2.js`:
- ADDED `Context().hide()`, `.show()`, `.swap()`, `.deleteElem()`, & `.createElem()`
- REMOVED `content` object and all references (including `contentStatus` property)
- REMOVED `userSaveButtonListenerExists`
- REARRANGED several methods and constructors, to make things easier to find
- RENAMED `App().roll` to `App().rollPress`
- RENAMED `App().toggleSaved` to `App().toggleMenu`
- ADDED `App()` functions:
- - `createSaveItem()`, `createSaveItemElement()` and `createSaveItemStorage()`
- - `deleteSaveItem()`, `deleteSaveItemElementAndListeners()` and `deleteSaveItemStorage()`
- - these work together to handle the creation and deletion of new save items
- ADDED `App().addUserSaveButtonListener()`
- ADDED `App()` functions: 
- - `userSaveButtonCheckDisplay()` which checks whether the userSaveButton should be hidden or visible
- - `userSaveButtonText()` which determines what text belongs in the userSaveButton

`index.html`:
- uncommented all page content, and added content for the savedMenu, so that everything except saved items are loaded from the start
- RENAMED `<button id='savedBtn'>` to `<button id='toggleMenuBtn'>`
- ADDED `<div>`s within the saveMenu, to target content written by script

Here's an [htmlPreview](http://htmlpreview.github.io/?https://github.com/EnderSmith/EnderSmith.github.io/blob/hideElements/diceRoll/index.html) as well.

### CODE REVIEW

File: diceRoll/diceRoll2.js [hunk 1]
```diff
@@ -82,56 +113,31 @@ function Listener(id, event, action, context) {
...
       this.addRollBarListeners();
-      //TODO: depricate this:
-      this.context.html('calcHolder', this.context.content.calculator, false);
       this.addCalculatorListeners();
+      this.addUserSaveButtonListener();
       this.simulateFirstVisit(true);
       this.checkMemory();
-    },
-
-    saveItemPress: function(id, rollArray) {
```
Comment (EnderSmith) [inline diceRoll/diceRoll2.js]: Not deleted, just moved

File: diceRoll/diceRoll2.js [hunk 2]
```diff
@@ -82,56 +113,31 @@ function Listener(id, event, action, context) {
   this.context.elem(id).addEventListener(this.event, this.exec.bind(this));
 }
 
-function Addend() {
```
Comment (EnderSmith) [inline diceRoll/diceRoll2.js:85]: Not deleted, just moved

File: diceRoll/diceRoll2.js [hunk 3]
```diff
@@ -158,104 +164,91 @@ function App(context) {
...
-      this.context.attach('savedBtn', 'click', this.toggleSaved.bind(this));
-    },
-    addSaveItemListeners: function() {
-      var saved = JSON.parse(this.context.storage().saved);
-      var saved_props = (Object.getOwnPropertyNames(saved));
-      for (var i = 0; i < saved_props.length; i++) {
-        var prop = saved[saved_props[i]];
-        var id = prop.id;
-        var rollArray = prop.rollArray;
-
```
Comment (EnderSmith) [inline diceRoll/diceRoll2.js:171]: These were refactored into a their own function, while the above became its own, further down

File: diceRoll/diceRoll2.js [hunk 4]
```diff
@@ -158,104 +164,91 @@ function App(context) {
...
-      savedMenu += "<div id='userSaveButton'>" +
-        "</div>" +
-        "</div>";
-
-      this.context.content.savedMenu = savedMenu;
-      //TODO: depricate this:
-      this.context.contentStatus = savedMenu;
-      return savedMenu;
     },
-    restoreDefaultSaveItems: function() {
```
Comment (EnderSmith) [inline diceRoll/diceRoll2.js:232]: Not deleted, just moved up

File: diceRoll/diceRoll2.js [hunk 5]
```diff
@@ -158,104 +164,91 @@ function App(context) {
...
+      this.context.createElem('div', rowId, 'row', 'saveItems');
+      this.context.createElem('button', id, 'btn saveItem col-m-8 col-t-8 col-8', rowId);
+      this.context.html(id, name + ': ' + this.sumArrayToDisplay(rollArray));
+      this.context.createElem('button', 'mod_' + id, 'btn saveItem col-m-2 col-t-2 col-2', rowId);
+      this.context.html('mod_' + id, 'mod');
+      this.context.createElem('button', 'delete_' + id, 'btn delete saveItem col-m-2 col-t-2 col-2', rowId);
+      this.context.html('delete_' + id, 'X');
     },
     deleteSaveItem: function(id) {
-      //TODO: depreicate:
```
Comment (EnderSmith) [inline diceRoll/diceRoll2.js:238]: Refactored into the three methods to the right

File: diceRoll/diceRoll2.js [hunk 6]
```diff
@@ -333,31 +326,23 @@ function App(context) {
     },
 
     // functions for displaying data
-    toggleSaved: function(override) {
```
Comment (EnderSmith) [inline diceRoll/diceRoll2.js:336]: This function was heavily overburdened and refactored to much more digestible chunks

File: diceRoll/diceRoll2.js [hunk 7]
```diff
@@ -367,25 +352,21 @@ function App(context) {
       var testOutput = [this.context.sumArray.length, this.context.sumIndex];
       return testOutput.toString();
     },
-    displayUserSaveButton: function() {
```
Comment (EnderSmith) [inline diceRoll/diceRoll2.js:370]: Renamed this function and factored it into the two on the right

**Review by EnderSmith — COMMENTED** [review]
Comments to clear things up, since it's so masive

**Review by ultimatetag — APPROVED** [review]
nice - hide/show of html is a huge improvement

**Pull request merged**